lib/netjockey/service.rb

```diff
@@ -62,16 +19,29 @@ def room(id)
       case res
       when Net::HTTPSuccess then
         data = JSON.parse(res.body)
-        room_info = RoomInfo.new(data['roomInfo']['name'],
-                                 data['roomInfo']['id'])
-        song_queue = data['queue']['playlist'].map do |song|
-          Song.new(song['duration'],
-                   song['title'], song['id'],
-                   song['thumbnailUrl'],
-                   song['uuid'])
-        end
-        Room.new(room_info, data['currentSongTime'], song_queue)
+        create_room(data)
+      when Net::HTTPNotFound then
+        raise 'room not found'
```

Comment:
By custom you mean that I should create new exception, or that I should use something like IOException?